docs/asciidoc/http-client.adoc

```diff
@@ -512,3 +518,188 @@ include::{examplesdir}/resolver/custom/Application.java[lines=18..38]
 ----
 <1> Sets the JVM built-in resolver.
 ====
+
+== Timeout Configuration
+This section describes various timeout configuration options that can be used in `HttpClient`.
+Configuring a proper timeout may improve or solve issues in the communication process.
+The configuration options can be grouped as follows:
+
+* <<connection-pool-timeout>>
+* <<http-client-timeout>>
+** <<response-timeout>>
+** <<connection-timeout>>
+** <<ssl-tls-timeout>>
+** <<proxy-timeout>>
+** <<dns-timeout>>
+
+[[connection-pool-timeout]]
+=== Connection Pool Timeout
+By default, `HttpClient` uses a connection pool. When a request is completed successfully and if the connection is not scheduled for closing,
+the connection is returned to the connection pool and thus it can be reused for processing another request. The connection may
```

Comment:
Change "thus it can be reused" to "can thus be reused". (Otherwise, you have to put commas around "thus".)